### DISCUSSION TITLE
Setup an IKEv2 client on macOS

### DISCUSSION THREAD

**Discussion by matteocarnelos** [strongswan/strongswan#1910] [Q&A] [answered]
Hi!

I'm trying to connect to an IKEv2 VPN server using `strongswan` on macOS 13.5.2 Ventura.

As suggested in the [documentation](https://docs.strongswan.org/docs/5.9/os/macos.html), I installed `strongswan` using Homebrew:
```
brew install strongswan
```
Then, I edited the `/usr/local/etc/ipsec.conf` and `/usr/local/etc/ipsec.secrets` files with all the details for the VPN connection, and finally established the connection by running:
```
sudo ipsec start
sudo ipsec up my-vpn
```
The connection gets correctly established, as I get:
```
connection 'my-vpn' established successfully
```
But unfortunately I cannot reach any host in the VPN.

By doing some research, it seems that I'm missing the network interface for the VPN connection (`ifconfig` doesn't return any new interface rather than the default macOS ones) and the corresponding routes in the routing table (by inspecting the routing table using `netstat -nr` there aren't any routes regarding the VPN subnet). I also tried to add them manually without any success.

Am I missing something to get the connection up and running? Unfortunately I couldn't find any help in the docs or other discussions.

Thanks for the help!

**Comment by tobiasbrunner** [comment]
> Am I missing something to get the connection up and running? Unfortunately I couldn't find any help in the docs or other discussions.

We obviously can't help you if you don't provide any useful information (see #196).

**Reply by matteocarnelos** [reply]
You're right, I'm sorry. I'm not an expert and I didn't want to accidentally leak sensitive information. However, I totally understand that I cannot ask for help without providing enough information. Therefore, I will stick to the working solution I found (posted here for reference) and further investigate it myself. Thank you for your help and time.

**Comment by matteocarnelos** [comment, marked as answer]
I managed to solve this issue by manually assigning the virtual IP to a free `utun` interface (e.g. `utun0`):
```
sudo ifconfig utun0 <virtual-ip>/<subnet> <virtual-ip>
```
Disabling the routing installation in `/usr/local/etc/strongswan.conf`:
```
charon {
    install_routes = no
    ...
}
```
Manually doing the routing:
```
sudo route add -net <remote-subnet> <virtual-ip>
```
And finally connecting to the VPN:
```
sudo ipsec start
sudo ipsec up my-vpn
```

However, I'm left with some questions:
1. Shouldn't strongSwan install the virtual IP? The setting `install_virtual_ip` of `strongswan.conf` suggests so.
2. If not disabled, the routing installation done by strongSwan seems to be done incorrectly (it routes on the wrong interface and IPs). Can this be related to #1807?

**Reply by tobiasbrunner** [reply]
>  1. Shouldn't strongSwan install the virtual IP?

Depends on whether the server assigns one etc.

>  2. If not disabled, the routing installation done by strongSwan seems to be done incorrectly (it routes on the wrong interface and IPs).

You have not provided any logs or configs so no idea what you are referring to.

> Can this be related to [Wrong route installed on FreeBSD? #1807](https://github.com/strongswan/strongswan/issues/1807)?

Who knows...

**Comment by barracuda156** [comment]
@tobiasbrunner Could you say if it needs `tuntapsox` as a dependency?

**Reply by tobiasbrunner** [reply]
I don't think so, why?

**Reply by barracuda156** [reply]
Some VPNs do. Anyway, I will try and see. Need to change the build config a bit in light of https://docs.strongswan.org/docs/latest/os/macos.html
I wanna get this running on macOS 10.5–10.6 on PowerPC (or at least 10.6).
